packages/inter-protocol/src/auction/auctionBook.js

```diff
@@ -89,6 +89,40 @@ export const makeBidSpecShape = (currencyBrand, collateralBrand) => {
 
 /** @typedef {import('@agoric/vat-data').Baggage} Baggage */
 
+/**
+ * @typedef {object} PriceLevel
+ * A description of a single quote
+ *
+ * @property {Amount<'nat'>} amountIn
+ * The amount supplied to a trade
+ * @property {Amount<'nat'>} amountOut
+ * The quoted result of trading `amountIn`
+ */
+
+/**
+ * @typedef {object} BookDataNotification
+ *
+ * @property {PriceQuote | null} startPrice identifies the priceAuthority and price
+ * @property {PriceLevel | null} currentPriceLevel the price at the current auction tier
+ * @property {Amount<'nat'> | null} startProceedsGoal The proceeds the sellers were targeting to raise
+ * @property {Amount<'nat'> | null} remainingProceedsGoal The remainder of the proceeds the sellers were targeting to raise
+ * @property {Amount<'nat'> | undefined} proceedsRaised The proceeds raised so far in the auction
+ * @property {Amount<'nat'>} startCollateral How much collateral was available
+ *    for sale at the start. (If more is deposited later, it'll be added in.)
+ * @property {Amount<'nat'> | undefined} collateralAvailable The amount of collateral remaining
+ */
+
+/*
+ * If stateVariable is unset, set it to amount, otherwise add amount
+ */
```

Comment:
`addAmountToState` sounds like it's adding a property to the state. this is _increasing_ an amount stored in state. The laziness of setting it if it isn't set seems unnecessarily dynamic. Why not set them to empty amounts during initState? 

Then you don't even need this function. The two places it's called can be like,
```js
state.remainingProceedsGoal = AmountMath.add(state.remainingProceedsGoal, incrementToGoal);
```
which is statically analyzable. If you really want to avoid repeating the state property we could have a helper like,
```js
increaseAmount(state, 'remainingProceedsGoal', incrementToGoal);
```
… and have the type checker verify that the second arg is a property in the first arg, but I think that still breaks things like "find all references".  

Kind of a general challenge with immutable data when you want to change it without repeating the name of the thing being changed.

Reply:
A couple of these values distinguish between empty and null.

Reply:
I removed the helper.  I considered renaming to `addToNullable()`, since state was the first arg anyway, but your comments about type checking and static analysis convinced me the function wasn't adding enough value.